### ISSUE TITLE
[iOS] Adding a MGLMapView in a storyboard crashes IB.

### ISSUE THREAD

**Issue by fabian-guerra** [mapbox/mapbox-gl-native#10573]
**Platform:** iOS 11
**Mapbox SDK version:** 3.7.0

### Steps to trigger behavior

 1. Add a `MGLMapView` through storyboards.

### Expected behavior
IB will work.

### Actual behavior
IB crashes with the following message:
```
.../Main.storyboard: error: IB Designables: Failed to render and update auto layout status for ViewController (BYZ-38-t0r): The agent crashed
```

This work well using `master` branch.

**Comment by dan-burt** [comment]
I am getting this message on a new iOS project that I have just started. How do I resolve? Are there some initial properties I need to set on the MGLMapView object?

**Comment by 1ec5** [comment]
Is this a duplicate of #10072?

**Comment by fabian-guerra** [comment]
You are right. Even tho I didn't use cocoapods it's the same error. Duplicate of #10072

**Comment by julianrex** [comment]
@fabian-guerra just want to confirm - you weren't using CocoaPods? 

cc @akitchen

**Comment by fabian-guerra** [comment]
@julianrex I confirm that I was not using cocoapods.

**Comment by 1ec5** [comment]
Reopening – #10072 is definitely caused by CocoaPods. This would be unrelated in that case. Do you have reliable steps to reproduce?

**Comment by fabian-guerra** [comment]
Import the SDK in a new app, or in the same test app storyboard add a map view. Has the same error as in #10072.

**Comment by 1ec5** [comment]
I can reproduce this issue in Xcode 9.4 with iOS map SDK v4.1.0-beta.1. With the “Developer” crash reporting preference turned on, I see this crash in IBDesignablesAgentCocoaTouch:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (Code Signature Invalid)
Exception Codes:       0x0000000000000032, 0x000000011d458000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x2
```

Same in Xcode 10 beta 1 with IBDesignablesAgent-iOS:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (Code Signature Invalid)
Exception Codes:       0x0000000000000032, 0x00000001262f7000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x2
```

The framework bundle isn’t codesigned. At build time, we rely on the “Code Sign On Copy” setting in the Embed Frameworks build phase. But apparently the designables agent now expects the bundle to be codesigned. I thought I’ve seen this work at times in the recent past, but perhaps I’m misremembering.

The codesigning issue explains why the macOS map SDK has never been affected by the designable crash, and why using a static framework can also avoid the crash.

In investigating this issue, I discovered that the designable agent has been renamed: #12140. That fix is necessary but insufficient for addressing the IB designable crash.

**Comment by tomoyuki28jp** [comment]
I cleaned my project and restarted my Mac, but I'm still facing the error.

**Comment by tomoyuki28jp** [comment]
FYI: subclassing MGLMapView solved it.

**Comment by friedbunny** [comment]
Fixed in #14379.
